src/qnetvo/network_ansatz.py

```diff
@@ -85,14 +113,12 @@ class NetworkAnsatz:
     measure layer is a collection of unitaries which encode the measurement basis.
     These layers are constructed from the ``prepare_nodes`` and ``measure_nodes`` inputs respectively.
 
-    :param prepare_nodes: A list of ``PrepareNode`` classes.
-    :type prepare_nodes: list[PrepareNode]
-
-    :param measure_nodes: A list of ``MeasureNode`` classes.
-    :type measure_nodes: list[MeasureNode]
-
-    :param noise_nodes: A list of ``NoiseNode`` classes.
-    :type noise_nodes: *optional* list[NoiseNode]
+    :param network_layers: Positional arguments each being a list of nodes designating a circuit layer.
+                           The first layer must contain ``PrepareNode`` classes while the last layer must
+                           contain ``MeasureNode`` classes.
+                           Intermediate layers can contain either ``NoiseNode`` or ``ProcessingNode`` classes,
```

Comment:
Is there much distinction between `NoiseNode` and `ProcessingNode`?

Reply:
The noise node is a processing node with 1 classical input. It would definitely be simpler to remove the `NoiseNode` class. The scope of breaking changes would grow, however. 

I left the class originally for its  "backwards compatability", but I'm inclined to think that maybe we should remove it since there are already breaking changes.